### PR TITLE
Extend stats topology by adding support for y-flow objects set

### PR DESCRIPTION
**Pull request by surabujin**
Inject y-flow objects into stats topology cache. Use new set of y-flow
object to porduce new metrics and new tags on existing metrics.

### CODE REVIEW

File: src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public void addInPortTag(long portNumber) {
+        tags.put("inPort", String.valueOf(portNumber));
+    }
+
+    public void addOutPortTag(long portNumber) {
+        tags.put("outPort", String.valueOf(portNumber));
+    }
+
+    public void addIsFlowSatelliteTag(boolean value) {
+        tags.put("is_flow_satellite", mapTagValue(value));
```
[the remark this answers is not in the document]
Comment (surabujin) [reply src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java:92]: I need verbose and independent way to define available range of values. At this moment the code is selfexplanation and this is exatly what I need.

File: src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    public void addFlowIdTag(String flowId) {
+        tags.put("flowid", flowId != null ? flowId : UNKNOWN_FLOW_ID_TAG_VALUE);
+    }
+
+    public void addYFlowIdTag(String yFlowId) {
+        tags.put("y_flow_id", yFlowId != null ? yFlowId : UNKNOWN_FLOW_ID_TAG_VALUE);
+    }
+
+    public void addDirectionTag(FlowPathDirection direction) {
```
[the remark this answers is not in the document]
Comment (surabujin) [reply src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java:51]: I have forgot to replace callers to use this shortcut. Nice catch. Fixed.

File: src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    public void addIsMirrorTag(boolean value) {
+        tags.put("is_mirror", mapTagValue(value));
+    }
+
+    public void addIsLoopTag(boolean value) {
+        tags.put("is_loop", mapTagValue(value));
+    }
+
+    public void addIsYFlowTag(boolean value) {
```
[the remark this answers is not in the document]
Comment (surabujin) [reply src-java/stats-topology/stats-storm-topology/src/main/java/org/openkilda/wfm/topology/stats/service/TagsFormatter.java:107]: Removed.

**Review by dpoltavets — APPROVED** [review]

File: src-java/stats-topology/stats-messaging/src/main/java/org/openkilda/messaging/info/stats/YFlowStatsInfoFactory.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    public UpdateYFlowStatsInfo produceAddUpdateNotification() {
+        return new UpdateYFlowStatsInfo(blank);
+    }
+
+    public RemoveYFlowStatsInfo produceRemoveYFlowStatsInfo() {
+        return new RemoveYFlowStatsInfo(blank);
+    }
+
+    private static class YFlowStatsInfoBlank extends BaseYFlowStatsInfo {
```
Comment (timofei-durakov) [inline src-java/stats-topology/stats-messaging/src/main/java/org/openkilda/messaging/info/stats/YFlowStatsInfoFactory.java:40]: Why do you name it blank? It is kind of prototype here, right?

**Review by timofei-durakov — APPROVED** [review]

File: src-java/stats-topology/stats-messaging/src/main/java/org/openkilda/messaging/info/stats/YFlowStatsInfoFactory.java (hunk 4 above)
Comment (surabujin) [reply src-java/stats-topology/stats-messaging/src/main/java/org/openkilda/messaging/info/stats/YFlowStatsInfoFactory.java:40]: Just following existing pattern `org.openkilda.floodlight.api.request.factory.IngressFlowSegmentRequestFactory`

**Pull request merged**